### DISCUSSION TITLE
Check if all nodes are connected

### DISCUSSION THREAD

**Discussion by khanarnab** [xyflow/xyflow#1694] [New Features]
Hello, I am trying to find a way to validate that my workflow is complete (the Start node is connected to the End node, directly or indirectly). Is there a way to validate that all of the nodes are connected. I would appreciate if any insight or advice.

**Comment by datoslabs** [comment]
The best way is to loop through the elements array and validate all incoming and outgoing edges for each element.  Each node type can have both required and optional connections much like a function; therefore, validation logic is specific to each implementation.
